### PR TITLE
Megadungeon

### PR DESCRIPTION
**Pull request by Khitiara**
Primary thing to investigate is how to do a boss room that spawns once per megadungeon. 
Also needs more rooms, custom blocks, and testing, once added to biome feature list.

Once complete, this pull request will fix issue #22

### CODE REVIEW

File: src/main/resources/assets/luminiferous_uplands/lang/en_us.json [hunk 1]
```diff
@@ -10,6 +10,7 @@
   "block.luminiferous_uplands.uplander_stone_brick_stairs": "Uplander Stone Brick Stairs",
   "block.luminiferous_uplands.uplander_stone_brick_slab": "Uplander Stone Brick Slab",
   "block.luminiferous_uplands.uplander_stone_brick_wall": "Uplander Stone Brick Wall",
+  "block.luminiferous_uplands.unbreakable_uplander_stone_bricks": "Unbreakable Uplander Stone Bricks",
```
Comment (Khitiara) [inline src/main/resources/assets/luminiferous_uplands/lang/en_us.json:18]: Any way to generalize this and have it use the translated name of the block it wraps, and use `Unbreakable %s` type thing?

Comment (kvverti) [reply src/main/resources/assets/luminiferous_uplands/lang/en_us.json:18]: Possible, but it's not as friendly for localization. Vanilla spawn eggs used to use generated localized names, but now they all have separate names.

File: src/main/java/robosky/ether/mixin/EntityContextImplMixin.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import robosky.ether.iface.BossDoorwayContext;
+
+@Mixin(EntityContextImpl.class)
+public abstract class EntityContextImplMixin implements BossDoorwayContext {
+
+    // it is very important that the lazy is only initialized after
+    // item registration. EntityContextImpl is loaded before Uplands registers
+    // its items, so a lazy field is needed.
+    @Unique
+    private static final Lazy<ItemStack> doorway =
```
Comment (Khitiara) [inline src/main/java/robosky/ether/mixin/EntityContextImplMixin.java]: Why does this need to be a field in particular?

Comment (kvverti) [reply src/main/java/robosky/ether/mixin/EntityContextImplMixin.java]: A new EntityContext is constructed potentially every tick. Constructing a new ItemStack every tick seems wasteful. I thought about using an item tag instead, what are your thoughts on that?

**Comment by kvverti** [comment]
Approved by @robotbrain and @devvoid

**Pull request merged**